draft-ietf-tls-external-psk-importer.md

```diff
@@ -79,156 +80,151 @@ when, and only when, they appear in all capitals, as shown here.
 
 # Overview
 
-Intuitively, key importers mirror the concept of key exporters in TLS in that they
-diversify a key based on some contextual information before use in a connection. In contrast to
-key exporters, wherein differentiation is done via an explicit label and context string,
-the key importer defined herein uses a label and set of hash algorithms to
-differentiate an external PSK into one or more PSKs for use.
+Key importers mirror the concept of key exporters in TLS in that they diversify a key
+based on some contextual information before use in a connection. In contrast to key exporters,
+wherein differentiation is done via an explicit label and context string, the key importer
+defined herein uses an optional context string along with a target protocol and KDF
+identifier to differentiate an external PSK into one or more PSKs for use.
 
 Imported keys do not require negotiation for use, as a client and server will not agree upon
 identities if not imported correctly. Thus, importers induce no protocol changes with
-the exception of expanding the set of PSK identities sent on the wire. Endpoints may incrementally
-deploy PSK importer support by offering non-imported keys for TLS versions prior to TLS 1.3.
-(Negotiation and use of imported PSKs requires both endpoints support the importer API described herein.)
+the exception of expanding the set of PSK identities sent on the wire. Endpoints may
+incrementally deploy PSK importer support by offering non-imported keys for TLS versions
+prior to TLS 1.3. (Negotiation and use of imported PSKs requires both endpoints support
+the importer API described herein.)
```

Comment:
I don't think that you need the parenthetical here any more.

Also, you should point out that imported PSKs and non-imported PSKs are distinguished by having different identifiers.  It seems obvious, but we know how little that matters when people are looking to play dumb.

Reply:
That seems reasonable. The parenthetical meant to make it painfully obvious that both sides need to import to use an external PSK. Replacing it with the suggested "non-imported and imported identities are different" text does the trick, too!